cordyceps/src/list.rs

```diff
@@ -264,6 +264,19 @@ pub struct IterMut<'list, T: Linked<Links<T>> + ?Sized> {
     len: usize,
 }
 
+/// An iterator returned by [`List::drain_filter`].
+pub struct DrainFilter<'list, T, F>
```

Comment:
Would it make sense to re-use `Cursor` here so we don't duplicate the iteration logic? You should be able to re-borrow the cursor mutable view into an immutable view for the predicate.